docs/table/construct_table.rst

```diff
@@ -471,6 +471,41 @@ To create a copy of an existing table that is empty (has no rows)::
  float64 float64
  ------- -------
 
+Empty array of a known size
+---------------------------
+
+If you do know the size your table will be, but don't know the values in
+advance, you can create a zeroed numpy array and build the astropy table from
+it::
+
+  >>> N = 4
+  >>> dtype = [('a', 'i4'), ('b', 'f8'), ('c', 'bool')]
+  >>> temp = np.zeros(N, dtype=dtype)
+  >>> t = Table(data=temp)
+  >>> t
+  <Table length=4>
+    a      b      c
+  int32 float64 bytes2
+  ----- ------- ------
+      0     0.0
+      0     0.0
+      0     0.0
+      0     0.0
+
+For example, you can then fill in this table row-by-row from extracted from another table, or generated on the fly::
+
+  >>> np.random.seed(5)
```

Comment:
I'm a bit wary on random numbers in docs, it's just begging for CI failures somewhere along the line.  What about something super-simple that does something but is trivial to understand:
```
>>> for i in range(len(t)):
...    t[i] = (i, 2.5 * i, i % 2)
>>> t
<Table length=3>
  a      b      c  
int32 float64  bool
----- ------- -----
    0     0.0 False
    1     2.5  True
    2     5.0 False
```